projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -152,36 +167,47 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     return (formValue as any) as ControlInterface;
   }
 
+  public onFormUpdate({ isInitialValue, formValue }: FormChange<FormInterface>): void {}
+
   public registerOnChange(fn: (_: any) => void): void {
     if (!this.formGroup) {
       return;
     }
 
     this.onChange = fn;
 
-    // this is required to correctly initialize the form value
-    // see note on-change-after-one-tick within the test file for more info
-    if (this.emitInitialValueOnInit) {
-      setTimeout(() => {
-        if (this.onChange && this.formGroup) {
-          this.onChange(this.transformFromFormGroup(this.formGroup.value));
-        }
-      }, 0);
-    }
-
     this.subscription = this.formGroup.valueChanges
       .pipe(
+        startWith(this.formGroupValues),
+        map((formValue, index) => ({ formValue, isInitialValue: index === 0 })),
         // this is required otherwise an `ExpressionChangedAfterItHasBeenCheckedError` will happen
         // this is due to the fact that parent component will define a given state for the form that might
         // be changed once the children are being initialized
         delay(0),
-        tap(changes => {
-          if (this.onTouched) {
-            this.onTouched();
+        filter(() => !!this.formGroup),
+        tap(({ formValue, isInitialValue }) => {
+          const formRemapValue = this.transformFromFormGroup(formValue);
+
+          if (!isInitialValue) {
+            if (this.onTouched) {
+              this.onTouched();
+            }
+          }
+
+          if (!isInitialValue || (isInitialValue && this.emitInitialValueOnInit)) {
+            if (this.onChange) {
+              this.onChange(formRemapValue);
+            }
           }
 
-          if (this.onChange) {
-            this.onChange(this.transformFromFormGroup(changes));
+          if (isNgxSubFormRemapComponent(this)) {
```

Comment:
I've created the `isNgxSubFormRemapComponent` here because:
1) it helps with type safety
2) I first tried to use directly `if (this instanceof NgxSubFormRemapComponent)` and it worked. But TsLint caught the fact that the class (`NgxSubFormRemapComponent`) cannot be used before its declaration, which is correct. Except that we target ES5 for building the lib which means that classes are transformed into functions which can then be correctly hoisted... So if anyone tries to use the library on their own and compiling into ES6, as classes are not hoisted it wouldn't work. Using the `isNgxSubFormRemapComponent` is a small trick for type safety + ES6 support (if I'm right) because when that check happens the `NgxSubFormRemapComponent` will be available.